includes/class-wc-stripe-api.php

```diff
@@ -242,6 +263,13 @@ public static function retrieve( $api ) {
 			]
 		);
 
+		// If we get a 401 error, we know the secret key is not valid, we save a transient to avoid making calls until the secrect key gets updated.
+		if ( ! empty( $response['response']['code'] ) && 401 === $response['response']['code'] ) {
+			$transient_key = WC_Stripe_Mode::is_test() ? self::TEST_MODE_INVALID_API_KEYS_TRANSIENT_KEY : self::LIVE_MODE_INVALID_API_KEYS_TRANSIENT_KEY;
+			set_transient( $transient_key, true );
```

Comment:
Do we want to do this immediately in all cases? Or only after we've triggered multiple failures in a short period of time?

I worry that there may be other situations where one or two 401 responses could occur -- I wouldn't want to stop a merchant's payments if that occurs.

Also, do we want to record the timestamp that we took the action? That may be useful when debugging and trying to understand what is going on.

Reply:
@diegocurbelo @daledupreez Did we address this concern somewhere? I wonder if one-off 401s are contributing to [this potential false positive issue](https://wordpress.org/support/topic/version-9-5-1-falsely-thinks-account-is-disconnected/) a user mentioned in forums

Reply:
API Keys can not be temporary disabled, only revoked; so getting one 401 means that we need new keys (there is no way for them to become valid again).

The user in the forum mentions that payments are still coming through, so we are not blocking API calls because of a 401 (the 401 flag is not set). It seems to have been a problem with the account cache; if the account can't be read (because of a timeout or a 500, 429 errors etc) the UI assumes the account is disconnected and show the notices.

We show refactor the account settings (and global notices) to account for the new flows we added to the Stripe API connector (which might cause the account/PM list to be null temporary), but that might be to big of a change for a patch release.

Reply:
That makes more sense. Thanks Diego! I see that it was resolved after clearing cache/transients.